modules/configuration/src/test/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtensionTest.java

```diff
@@ -169,4 +174,19 @@ void testParamConfigurationStorageRevisionListenerHolder(
 
         assertTrue(revisions.get(0) < revisions.get(1), revisions::toString);
     }
+
+    /** Test UUID generation in mocks. */
+    @Test
+    public void testInjectInternalId(
+            @InjectConfiguration(
+                    name = "test",
+                    polymorphicExtensions = {
+                            SortedIndexConfigurationSchema.class,
+                            UnknownDataStorageConfigurationSchema.class
+                    },
+                    internalExtensions = { ExtendedTableConfigurationSchema.class }
+            ) TableConfiguration tableCfg
```

Comment:
Let's, by analogy with **org.apache.ignite.internal.configuration.schema.ExtendedTableConfigurationSchema**, create an inheritor from **org.apache.ignite.internal.configuration.sample.DiscoveryConfigurationSchema** with the field
```
    @InternalId
    public UUID id;
```

Reply:
But why not to use `ExtendedTableConfigurationSchema`?

Reply:
Module **ignite-extended-api** must depend on **ignite-configuration** and not vice versa.